src/literals.py

```diff
@@ -173,6 +173,15 @@ def __eq__(self, value: object, /) -> bool:
 ]
 
 
+class _RebalanceMode:
+    FULL = "full"
+    ADD = "add"
+    REMOVE = "remove"
+
+
+RebalanceMode = _RebalanceMode()
```

Comment:
**todo:** Use a `TypeAlias` here? What is this private class doing?

Reply:
**opinion:** I also really like Literals rather than just doing `Class.UPPERCASE` for a lower-case string.

Reply:
The problem with a literal type is that I cannot simply access one of its members for equality. An earlier commit used this pattern

```py
from typing import Literal, get_args

RebalanceMode: TypeAlias = Literal["full", "add", "remove"]

if mode != get_args(RebalanceMode)[0]:
    pass

```

But I could totally have a mix of both with something like that:
```py

MODE_FULL = "full"
...
RebalanceMode: TypeAlias = Literal[MODE_FULL, MODE_ADD, MODE_REMOVE]
```

Would you prefer that?

Reply:
I do! And yeah, `get_args` smells bad, agreed.